### ISSUE TITLE
vectorize digest

### ISSUE THREAD

**Issue by ehartford** [eddelbuettel/digest#59]
Can you please consider vectorizing digest?

https://stackoverflow.com/questions/42335787/why-isnt-mutate-working
https://stackoverflow.com/questions/28358549/why-does-the-digest-function-return-the-same-value-every-time-when-used-with-dpl
http://stackoverflow.com/questions/32465074/new-dataframe-column-as-function-digest-of-another-one-is-not-working-for-me

**Comment by eddelbuettel** [comment]
Awesome.

Where in the hypothetical "open source contract" did it say that I am the only one working on this, and/or that you could ask me to do pro bono work for your pleasure?

So with your that: _can you please *grow up* and learn that if you want change to happen FAST you need to be prepared to do it yourself?_  We tend to call that pull requests around here.

If the digest package does not suit your pristine and elaborate needs, you are always welcome to exercise your money-back guarantee,

**Comment by ehartford** [comment]
I asked nicely.  If the answer is 'no' then there's nothing wrong with that.  Sorry you are stressed.

**Comment by eddelbuettel** [comment]
That's the point: you didn't ask "nicely".  You piled up a bunch of references you clearly did not understand yourself.  Mr Flick is spot on: `digest()` is designed to return _one_ hash digest for _the whole object_ and he even showed you how to (trivially) create a vectorize version by calling, wait for it, the base R function `Vectorize()`:

```r
R> library(digest)
R> vdig <- Vectorize(digest)
R> vdig(1:5)
[1] "4b5630ee914e848e8d07221556b0a2fb" "c01f179e4b57ab8bd9de309e6d576c48" "11946e7a3ed5e1776e81c0f0ecd383d0"
[4] "234a2a5581872457b9fe1187d1616b13" "dd4ad37ee474732a009111e3456e7ed7"
R> 
```

**Comment by ehartford** [comment]
sorry.

**Comment by eddelbuettel** [comment]
No worries.  digest was made for simple hash checksums of R objects.  It use grew.

If you need blazing fast hashes of many (and large) object, then [fastdigest](https://cran.r-project.org/package=fastdigest).  I still think it is a "bug" in their package that they didn't contribute it to digest (as a dozen other authors have over the decade) but such is life ...  Can't win'em all.

**Comment by wlandau** [comment]
Maybe it's not to late. @gmbecker, what do you think? Integration of `fastdigest` into `digest` would automatically solve https://github.com/richfitz/storr/issues/82 and speed up [`drake`](https://github.com/ropensci/drake).

**Comment by wlandau** [comment]
Relevant: https://github.com/gmbecker/fastdigest/issues/2

**Comment by eddelbuettel** [comment]
I wrote that in 2017, and I stand by it.  The R community would be better off if we made representative packages better, rather than going off in our corner cases.  But that's how it goes, and why we now have three benchmarking packages, two digesting packages, two unit testing frameworks, several packages for XML reading, several for JSON, three (or four if you count a withdrawn one)  for PostgreSQL access and on and on and on. 

In each case a microargument can be, has been made, is being made. See your last comment here. In the long run it still sucks.  Because we're all worse off.

`digest` was my second package. I wrote it starting in 2003 or 2004.  Are there designs in it that could be better?  Sure.  John Chambers even roasts for me it his 2008 Software for Data Analysis book.  

Does that excuse us (collectively) from not trying to make it better?  I'd say, you say yes, and Will suffers for having a worse accessor package.  So be it.  There is only a limited number of battles each one of us can fight.
